### DISCUSSION TITLE
Salt ssl.init error

### DISCUSSION THREAD

**Discussion by LS3-Allen** [Security-Onion-Solutions/securityonion#5274] [Unsupported Versions] [answered]
v2.3.70
Distributed deployment
1 manager + 3 heavy nodes
`Checking Docker status

    Docker ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]

Checking container statuses

    so-aptcacherng --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-cortex -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-dockerregistry ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ [ OK ]
    so-elastalert ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-elasticsearch ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-filebeat ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ [ OK ]
    so-fleet --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-grafana ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-idstools ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ [ OK ]
    so-influxdb ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ [ OK ]
    so-kibana -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-kratos -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-logstash ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ [ OK ]
    so-mysql --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-nginx --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-playbook ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ [ OK ]
    so-redis --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-sensoroni ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-soc ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-soctopus ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ [ OK ]
    so-telegraf ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ [ OK ]
    so-thehive ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-thehive-es ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-wazuh --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
`

**Comment by LS3-Allen** [comment]
I am having trouble adding Osquery agents to my deployment.  I ran `sudo systemctl status launcher.so-launcher.service -l` and get the following error.
`authentication handshake failed: x509: certificate is valid for 10.2.1.100, not 10.2.10.100`

The manager's IP address is 10.2.10.100 and I have double checked global.sls and manager.sls to make sure everything is correct.  I'm not sure where the 10.2.1.100 IP address came from.  I did some reading on this forum and found a suggestion to recreate the SSL certificate using the `sudo salt-call state.apply ssl.init` command.  I tried that but get an error saying:

> [ERROR   ] Failure!
> local:
> ----------
>           ID: ssl.init_state_not_allowed
>     Function: test.fail_without_changes
>       Result: False
>      Comment: Failure!
>      Started: 20:58:34.439833
>     Duration: 2.239 ms
>      Changes:
> 
> Summary for local
> ------------
> Succeeded: 0
> Failed:    1
> ------------
> Total states run:     1
> Total run time:   2.239 ms

**Reply by LS3-Allen** [reply]
How do I get around the ssl.init state not allowed issue?

**Comment by LS3-Allen** [comment, marked as answer]
I never did figure out the ssl.init error, but I did fix the SSL certificate handshake error.

The problem was when I initially ran setup I accidentally had a typo in the IP address that I did not catch until after setup was finished.  When I realized the IP address was wrong I simply ran setup again.  Everything turned out good after the second setup, however the etc/pki directory still had the certificates from the first setup.

For whatever reason I never could get sudo salt-call state.apply ssl.init to work.  I had to end up deleting the .crt and .key files in etc/pki then run setup for a third time.  This time the SSL certificates were created with the correct IP address.

**Comment by SuiNom** [comment]
Hey Folks,

Just wanted to add to this for anyone else coming after me. I was encountering the same error every time I tried to regenerate the certificates after updating `fleet_custom_hostname' in global.sls:
```
[root@security-onion-0 securityonion]# salt-call state.apply ssl.init
[ERROR   ] Failure!
local:
----------
          ID: ssl.init_state_not_allowed
    Function: test.fail_without_changes
      Result: False
     Comment: Failure!
     Started: 16:01:15.396239
    Duration: 2.342 ms
     Changes:   

Summary for local
------------
Succeeded: 0
Failed:    1
------------
Total states run:     1
Total run time:   2.342 ms
```

This was after following all the instructions from [this post](https://github.com/Security-Onion-Solutions/securityonion/discussions/1530#discussioncomment-134906). I was a little frustrated and wasn't making any progress, so I decided to go rouge and take the following steps:

1. Snapshot my Standalone Manager - Just in case...
2. Backup the relevant certificates I thought I was most interested in tinkering with to the default onion users home dir:
```
cp /etc/pki/fleet.* /home/onion/
cp /etc/pki/managerssl.* /home/onion/
```
3. Blow away those certificates:
```
rm /etc/pki/fleet.*
rm /etc/pki/managerssl.*
```
4. Reboot: `reboot`
5. SSH back in and patiently wait for all the services to start `sudo watch -cd -n 10 -d so-status`
6. Check the `/etc/pki/` folder for the fleet and managerssl certs and keys - I was half expecting nothing, but to my surprise they'd been regenerated.
7. Check that the certificates included my `fleet_custom_hostname` as a Subject Alternate Name `openssl x509 -in /etc/pki/managerssl.crt -text -noout`

Success!

After this I just restarted my osquery `launcher` service on my test box and it connected in no problem ([I'd already done all the pre-reqs](https://docs.securityonion.net/en/2.3/osquery.html)) and regenerated the packages prior as part of my other troubleshooting steps.

Hopefully this saves someone a day of pain, cheers!

**Reply by LS3-Allen** [reply]
I ended up deleting everything in the /etc directory and then doing a reinstall on the manager and all nodes.  I didn't even think to check and see if the certificates where being auto generated.  I'm sure @SuiNom's process would have saved a bunch of trouble and time.  Thank you!
